core/src/main/java/com/datastrato/gravitino/catalog/CatalogOperationDispatcher.java

```diff
@@ -436,12 +439,29 @@ public Table createTable(
     NameIdentifier catalogIdent = getCatalogIdentifier(ident);
     doWithCatalog(
         catalogIdent,
-        c ->
-            c.doWithPropertiesMeta(
-                p -> {
-                  p.tablePropertiesMetadata().validatePropertyForCreate(properties);
-                  return null;
-                }),
+        c -> {
+          Set<TableCatalogInability> tableCatalogInabilities =
+              c.doWithTableOps(TableCatalog::inability);
+          Optional<TableCatalogInability> columnDefaultInability =
+              tableCatalogInabilities.stream()
+                  .filter(
+                      i -> i.isCapability(TableCatalogInability.Capability.COLUMN_DEFAULT_VALUE))
+                  .findAny();
+          columnDefaultInability.ifPresent(
+              i -> {
+                for (Column column : columns) {
+                  Preconditions.checkArgument(
+                      Column.DEFAULT_VALUE_NOT_SET.equals(column.defaultValue()),
+                      i.unsupportedReason() + " Illegal column " + column.name());
+                }
+              });
+          c.doWithPropertiesMeta(
+              p -> {
+                p.tablePropertiesMetadata().validatePropertyForCreate(properties);
+                return null;
+              });
+          return null;
+        },
```

Comment:
Alter table operation also need to check

Reply:
What needs to be checked? Alter table change doesn't support altering column default value